bin/create_juno_qc_report.py

```diff
@@ -0,0 +1,269 @@
+sys.stdout = sys.stderr = open(snakemake.log[0], "w")  # type: ignore
+
+import pandas as pd
+from functools import reduce
+import json
+
+
+def get_genus(species_csv: str) -> pd.DataFrame:
+    """
+    Get the metadata file and create a pandas dataframe
+    """
+    df = pd.read_csv(species_csv, usecols=["sample", "genus"])
+    df["sample"] = df["sample"].astype(str)
+    return df
+
+
+def get_phred_score(phred_json: str) -> pd.DataFrame:
+    """
+    Parses multiqc_data json file to get phred quality scores for each sample into a pandas dataframe
+    """
+    with open(phred_json) as f:
+        phred_data = json.load(f)
+
+    phred = phred_data["report_plot_data"]["fastqc_per_sequence_quality_scores_plot"][
+        "datasets"
+    ]
+
+    phred2 = []
+    for val in phred:
+        for i in val:
+            phred2.append(i)
+
+    df = pd.DataFrame(phred2)
+    parsed_df = (
+        df.groupby("name").data.apply(lambda x: pd.DataFrame(x.values[0])).reset_index()
+    )
+
+    parsed2_df = parsed_df.drop("level_1", axis=1)
+    parsed2_df.rename(columns={0: "phred", 1: "Sequences"}, inplace=True)
+    # Remove reads containing _pR
+    parsed3_df = parsed2_df[~parsed2_df["name"].str.contains("_pR")]
+
+    # get phred score by taking the score associated with the highest number of sequences per read
+    parsed4_df = parsed3_df.groupby("name")["Sequences"].max().reset_index()
+    parsed5_df = parsed3_df.merge(parsed4_df)
+    parsed5_df["name"] = parsed5_df["name"].astype(str).apply(lambda x: x.split("_")[0])  # type: ignore
+
+    # set mean phred score of R1 and R2
+    parsed6_df = parsed5_df.groupby("name")["phred"].mean().reset_index()
+    # rename name to sample to match rest of outputs
+    parsed6_df.rename(columns={"name": "sample"}, inplace=True)
+    parsed6_df["sample"] = parsed6_df["sample"].astype(str)
+
+    return parsed6_df
+
+
+def get_sequence_len(seq_len_tsv: str) -> pd.DataFrame:
+    """
+    Get the average sequence length from multiqc_fastqc.txt
+    """
+    df = pd.read_csv(seq_len_tsv, sep="\t", usecols=["Sample", "avg_sequence_length"])
+    df.rename(columns={"Sample": "sample"}, inplace=True)
+    # remove reads containing _pR
+    df = df[~df["sample"].str.contains("_pR")]
+
+    # sample name to match metadata
+    df["sample"] = df["sample"].apply(lambda x: x.split("_")[0])  # type: ignore
+
+    # average sequence length of R1 and R2
+    df = df.groupby("sample")["avg_sequence_length"].mean().reset_index()
+    df["avg_sequence_length"] = df["avg_sequence_length"].round(decimals=0)
+    df["sample"] = df["sample"].astype(str)
+
+    return df
+
+
+def get_transposed_report(quast_tsv: str) -> pd.DataFrame:
+    """
+    Creates a dataframe with necessary columns from the quast transposed report
+    """
+    df = pd.read_csv(
+        quast_tsv,
+        sep="\t",
+        usecols=["Assembly", "Total length", "# contigs", "N50", "GC (%)"],
```

Comment:
Maybe it makes sense to add units to Total length (Mbp), completeness (%) and  contamination (%) in the output excel

Reply:
Agreed, will do so